### PR TITLE
implementation modelmapper in laundry service

### PR DESCRIPTION
**Pull request by RusskovMatvey**

### CODE REVIEW

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+@Configuration
+public class OrderServiceBeansConfig {
+
+    @Bean
+    ModelMapper modelMapper() {
+        ModelMapper modelMapper = new ModelMapper();
+        modelMapper.getConfiguration()
+                .setFieldMatchingEnabled(true)
+                .setFieldAccessLevel(PRIVATE);
```
Comment (shinehlina) [inline laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: Do you really need those configs if you use Converter?

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                destination.setId(source.getOrderId());
+                destination.setSubmittedTime(System.currentTimeMillis());
+                destination.setEstimatedTime(source.getDetails().stream()
+                        .map(DetailSubmissionDto::getDuration)
+                        .reduce(0L, Long::sum));
+                destination.setStatus(OrderStatus.QUEUED);
+                destination.setLaundryId(source.getLaundryId());
+                destination.setBucket(source.getBucket());
+                return destination;
+            }
```
Comment (shinehlina) [inline laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: In this implementation model Mapper doesn't do anything.
You can probably extract some logic into empty constructor (like order status and submittedTime), use typeMap for non standard field (like setId) and postConverter. (I believe that some mapping can be performed automatically)

**Review by shinehlina — CHANGES_REQUESTED** [review]

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java (hunk 2 above)
Comment (AfanasievYuriy) [reply laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: > In this implementation model Mapper doesn't do anything.
> You can probably extract some logic into empty constructor (like order status and submittedTime), use typeMap for non standard field (like setId) and postConverter. (I believe that some mapping can be performed automatically)

please, provide examples (your own code or links to articles/guides) for @RusskovMatvey to learn

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java (hunk 1 above)
Comment (RusskovMatvey) [reply laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: You are right, this config is not needed.

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java (hunk 2 above)
Comment (RusskovMatvey) [reply laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: I fixed. May you check?

File: laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        return modelMapper;
+    }
+
+    private Converter<OrderSubmissionDto, OrderEntity> toOrderEntityConverter = context -> {
+        OrderSubmissionDto source = context.getSource();
+        OrderEntity destination = context.getDestination();
+
+        destination.setSubmittedTime(System.currentTimeMillis());
+        destination.setEstimatedTime(source.getDetails().stream()
```
Comment (RusskovMatvey) [inline laundry-management-service/laundrymanagement-app/src/main/java/com/microservices/laundrymanagement/configuration/OrderServiceBeansConfig.java]: I tried to put this (setEstimatedTime) out in the configure, but ConfigurationException and NullPointerException crashed

**Review by shinehlina — APPROVED** [review]

**Pull request merged**